packages/network-controller/src/NetworkController.ts

```diff
@@ -293,11 +295,22 @@ export class NetworkController extends BaseControllerV2<
         this.#setupInfuraProvider(type);
         break;
       case NetworkType.localhost:
-        this.#setupStandardProvider(LOCALHOST_RPC_URL);
+        if (chainId === undefined) {
+          throw new Error('chainId must be provided for custom RPC endpoints');
+        }
+        this.#setupStandardProvider(LOCALHOST_RPC_URL, toHex(chainId));
         break;
       case NetworkType.rpc:
-        rpcTarget &&
-          this.#setupStandardProvider(rpcTarget, chainId, ticker, nickname);
+        if (chainId === undefined) {
+          throw new Error('chainId must be provided for custom RPC endpoints');
+        }
+
+        if (rpcTarget === undefined) {
```

Comment:
The check for `rpcTarget` was added because `createNetworkClient` needs one in order to pass it along to the `fetch` middleware.